### ISSUE TITLE
[BUG] set_values() cannot handle Label types

### ISSUE THREAD

**Issue by brimoor** [voxel51/fiftyone#1919]
On `fiftyone==0.16.5`, an error is raised when trying to use `set_values()` to set frame-level `Label` fields.

For example, this works on `fiftyone==0.15.1` but currently raises an error:

```py
import fiftyone as fo

sample = fo.Sample(filepath="video.mp4")

dataset = fo.Dataset()
dataset.add_sample(sample)

labels = [{1: fo.Classification(label="cat")}]
dataset.set_values("frames.classification", labels)
```

```
InvalidDocument: cannot encode object: <Classification: {
    'id': '62bf7d8a1dc41887c27b809a',
    'tags': BaseList([]),
    'label': 'cat',
    'confidence': None,
    'logits': None,
}>, of type: <class 'fiftyone.core.labels.Classification'>
```
